blip-00xx.md

```diff
@@ -0,0 +1,220 @@
+```
+bLIP: xx 
+Title: BOLT 11 Invoice Blinded Path Tagged Field
+Author: Elle Mouton <elle.mouton@gmail.com>
+Status: Draft
+Created: 2024-06-20
+* Post-History: dates of postings to lightning-dev mailing list, or link to thread in 
+  mailing list archive
+License: CC0
+```
+
+## Abstract
+
+This bLIP defines a new [tagged field][tagged-fields] for the payment invoice
+encoding described by [BOLT 11][bolt11] which can be used to communicate 
+encoded blinded path information to the payer of the invoice. 
+
+## Copyright
+
+This bLIP is licensed under the CC0 license.
+
+## Rational
+
+Blinded paths have been included in the [BOLT 4 specification][blinded-paths] 
+and using them in the context of receiving payments is natively supported in 
+the [BOLT 12 Offers proposal][offers]. However, the Offers proposal also 
+includes various other dependencies such as onion messaging and various new 
+protocol message all of which will require a network wide upgrade before full 
+advantage can be taken of the new protocol. Blinded paths themselves are a 
+useful tool for privacy and potentially more reliable payment delivery and so 
+this document proposes a carve-out to the existing [BOLT 11] invoice format so 
+that advantage can be taken of the blinded paths protocol in the context of 
+payments in a sooner time frame. This will be done by adding a new tagged-field 
+type to the BOLT 11 invoice specification that will encode a blinded payment 
+path. With this bLIP, the sender and the receiver of a payment will need to be 
+aware of the new tagged-field and the receiver will need to support route 
+blinding and ideally have direct channel peers who also support route blinding 
+in order to start widening their anonymity set.
+
+## Specification
+
+### Tagged Field
+
+The proposal is to add the following new [tagged field][tagged-fields] to the 
+set defined in [BOLT 11][bolt11]:
+
+- `b` (20): `data_length` variable. One or more entries each containing a 
+   blinded payment path for a private route; there may be more than one `b` 
+   field. The field uses the `blinded_payinfo` type described below which draws 
+   heavily on the proposed encoding of the `blinded_payinfo` subtype defined in 
+   the [Offers proposal][offers].
+
+1. subtype: `blinded_payinfo`
+2. data:
+  * [`u32`:`fee_base_msat`]
+  * [`u32`:`fee_proportional_millionths`]
+  * [`u16`:`cltv_expiry_delta`]
+  * [`u64`:`htlc_minimum_msat`]
+  * [`u64`:`htlc_maximum_msat`]
+  * [`u16`:`flen`]
+  * [`flen*byte`:`features`]
+  * [`33*byte`:`first_ephemeral_blinding_point`]
+  * [`byte`:`num_hops`]
+  * [`num_hops*blinded_hop`:`blinded_hops`]
+
+1. subtype: `blinded_hop`
+2. data:
+   * [`33*byte`:`blinded_node_pubkey`]
+   * [`2*byte`:`cipher_text_length`]
+   * [`cipher_text_length*byte`:`cipher_text`]
+
+The `blinded_node_pubkey` of the first `blinded_hop` in a `blinded_payinfo` is
```

Comment:
Why do we need to do it this way, meaning that although the field is called blinded, we use the real pubkey ?

Reply:
it's just for the very first hop (the intro node) and is done as a way to save space. Else we would need a whole separate 33 byte field for the intro node pub key. The intro node's blinded pub key is never used. 

I can add this explanation though if it is unclear 👍 

The other option is of course just special casing the first hop's encoding and letting it only contain cipher text but I opted for a simpler encoding here

Reply:
I think carving out the intro node's real key into its own field in the root `blinded_payinfo` structure is a better way to organize this so that we have full homogeneity within the `blinded_hop` vector. I don't think space is saved either way since any data that we include in the main message wouldn't be included in `blinded_hops`.

Having exceptions, asterisks, or other caveats makes the spec harder to implement correctly.

Reply:
> I don't think space is saved either way

space saved in the invoice itself though :) ie trying to make it still fit in a qr code. 

but yeah - can change this to be more explicit if enough people agree. Just seems strange to include the intro node's blinded key when that is never used

Reply:
No I mean why not omit the intro node's "hop" from `blinded_hops` completely, and then just include the intro node's ciphertext in its own field?

Reply:
> No I mean why not omit the intro node's "hop" from blinded_hops completely, and then just include the intro node's ciphertext in its own field?

IIUC it's done this way to mirror the Offers encoding.

Reply:
ah ok i see what you mean now, like a separate `intro_hop` and `intro_encrypted_data`. 

The only arg i see against this is that then the num_hops isnt really the true number of hops. But yeah happy to change this. I think what i'll do is leave it as is for now and then will add this note to the "discussions/questions" section for discussion.